dotnet/src/Connectors/Connectors.MistralAI/MistralAIServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Net.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Connectors.MistralAI;
+using Microsoft.SemanticKernel.Embeddings;
+using Microsoft.SemanticKernel.Http;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Provides extension methods for the <see cref="IServiceCollection"/> interface to configure Mistral connectors.
+/// </summary>
+public static class MistralAIServiceCollectionExtensions
+{
+    /// <summary>
+    /// Adds an Mistral chat completion service with the specified configuration.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> instance to augment.</param>
+    /// <param name="model">The name of the Mistral model.</param>
+    /// <param name="apiKey">The API key required for accessing the Mistral service.</param>
+    /// <param name="endpoint">Optional  uri endpoint including the port where MistralAI server is hosted. Default is https://api.mistral.ai.</param>
+    /// <param name="serviceId">A local identifier for the given AI service.</param>
+    /// <param name="httpClient">The HttpClient to use with this service.</param>
+    /// <returns>The same instance as <paramref name="services"/>.</returns>
+    public static IServiceCollection AddMistralChatCompletion(
+        this IServiceCollection services,
+        string model,
+        string apiKey,
+        Uri? endpoint = null,
+        string? serviceId = null,
+        HttpClient? httpClient = null)
```

Comment:
There were a few issues, such as https://github.com/microsoft/semantic-kernel/issues/5065 and https://github.com/microsoft/semantic-kernel/issues/4950, regarding having httpClient as a parameter for the `IServiceCollection.Add*` extension methods. Both were closed by design. On the other hand, I can see that several Add* extension methods for the service collection accept the client. Let's agree on and stick to one approach.